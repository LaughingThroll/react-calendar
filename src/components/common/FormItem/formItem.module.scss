.formItem {
  &__title {
    margin-bottom: 6px;
  }
  &__inner {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  input {
    max-width: 140px;
    width: 100%;

    & + input {
      margin-left: 30px;
    }
  }
}
