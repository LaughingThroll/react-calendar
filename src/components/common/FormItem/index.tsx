import React from 'react'
import styles from './formItem.module.scss'
import { TChildren } from '../../../types/utilsTypes'

export interface FormItemProps {
  title?: string
  inner?: boolean
  children?: TChildren
  classNames?: string[]
}

const FormItem: React.FC<FormItemProps> = ({ title = 'Title', inner, classNames = [], children }) => {
  const { formItem, formItem__title, formItem__inner } = styles

  return (
    <div className={[formItem, ...classNames].join(' ')}>
      <h5 className={formItem__title}>{title}</h5>
      {inner ? <div className={formItem__inner}> {children}</div> : children}
    </div>
  )
}

export default React.memo(FormItem)
