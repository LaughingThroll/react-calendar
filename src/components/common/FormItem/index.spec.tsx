import React from 'react'
import { shallow } from 'enzyme'
import FormItem, { FormItemProps } from '.'

const setUp = (props: FormItemProps) => shallow(<FormItem {...props} />)

describe('FormItem render', () => {
  it('default render', () => {
    const tree = setUp({})

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes('Title')).toBe(true)
    expect(tree.find('.formItem__inner')).toHaveLength(0)
  })
})

describe('FormItem props', () => {
  it('props "title"', () => {
    const tree = setUp({ title: 'String' })

    expect(tree.text().includes('String')).toBe(true)
  })

  it('props "inner"', () => {
    const tree = setUp({ inner: true })

    expect(tree).toMatchSnapshot()
    expect(tree.find('.formItem__inner')).toHaveLength(1)
  })
})
