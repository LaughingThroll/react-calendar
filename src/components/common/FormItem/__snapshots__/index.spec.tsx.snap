// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem props props "inner" 1`] = `
<div
  className="formItem"
>
  <h5
    className="formItem__title"
  >
    Title
  </h5>
  <div
    className="formItem__inner"
  >
     
  </div>
</div>
`;

exports[`FormItem render default render 1`] = `
<div
  className="formItem"
>
  <h5
    className="formItem__title"
  >
    Title
  </h5>
</div>
`;
